Convert css nano to conditional include

michealengland opened the pull request:
Does not have open issue, must be tested locally.

### DESCRIPTION

This feature improves performance when using `npm run start`. By preventing `css-nano` from running in development, rebuild times are quicker due to less processing needed.

### OTHER

- [NA] Is this issue accessible? (Section 508/WCAG 2.0AA)
- [x] Does this issue pass all the linting? (PHPCS, ESLint, SassLint)
- [NA] Does this pass CBT?

### STEPS TO VERIFY

1. Before checking out this branch take an inventory of build times when using `npm run start` on `main`. Set steps below for details on rebuild.
1. Next, checkout this `feature/improve-build-tool-speed` and run `npm run start`. 
1. Once the script has built, edit an `@apply` css statement. Simply commenting out files may give false positives as these changes are incredibly fast regardless.
1. Verify that your `build/style-index` is unminified when using `npm run start`. 
1. Verify that your `build/style-index` is minified when using `npm run build`. 

### DOCUMENTATION

No additional documentation is required.

Commit:
Convert css nano to conditional include

bradp commented:
This could be simplified to be:
```
module.exports = {
	plugins: [
		require("tailwindcss"),
		require("autoprefixer"),
		process.env.NODE_ENV === "production" ? cssnano({ preset: "default" }) : null,
	],
};
```

michealengland commented:
> This could be simplified to be:
> 
> ```
> module.exports = {
> 	plugins: [
> 		require("tailwindcss"),
> 		require("autoprefixer"),
> 		process.env.NODE_ENV === "production" ? cssnano({ preset: "default" }) : null,
> 	],
> };
> ```

@bradp I like the simplification, but I also did some research on this and think we maybe should deviate slightly from what you have shown here.

Using `process.env.NODE_ENV === "production" ? cssnano({ preset: "default" }) : null` results in a `null` value left behind and using `process.env.NODE_ENV && "production" ? cssnano({ preset: "default" }) ` results in a falsey value left behind. From what I'm seeing neither of these values should be passed into the Webpack plugins, see https://github.com/webpack/webpack/issues/5493#issuecomment-321730266.

As an alternative, maybe we should follow the example seen here and fully remove false values from the plugins https://github.com/webpack/webpack/issues/5493#issuecomment-321705298.

What are your thoughts? On a side note, I did a test and passing `null` did work as expected but it might be good to error on the side of caution.

bradp commented:
I'd say 👍  to running it through the `.filter(Boolean)`

My main goal is to keep it more readable and in one place, so that works for me.

Commit:
Simplify conditional

Since the ternary operator leaves behind a false value, we should filter
this out since it is unwanted.

This update is based on https://github.com/webpack/webpack/issues/5493

coreymcollins reviewed (approved):
Nice, this does bring about a nice little boost! On `main`, `npm run start` took about 20 seconds and in this branch it's around 15. It also dropped the compile time while running `npm run start` by about 1-1.5 seconds, fluctuating between the two.

Thanks!
